src/main/java/com/hazelcast/kubernetes/DnsEndpointResolver.java

```diff
@@ -85,16 +86,18 @@ private static DirContext createDirContext(int serviceDnsTimeout) {
 
     private List<DiscoveryNode> lookup()
             throws NamingException, UnknownHostException {
-        Attributes attributes = dirContext.getAttributes(serviceDns, new String[]{"SRV"});
-        NamingEnumeration<?> servers = attributes.get("srv").getAll();
-
         Set<String> addresses = new HashSet<String>();
-        while (servers.hasMore()) {
-            String server = (String) servers.next();
-            String serverHost = extractHost(server);
-            InetAddress address = InetAddress.getByName(serverHost);
-            if (addresses.add(address.getHostAddress()) && logger.isFinestEnabled()) {
-                logger.finest("Found node service with address: " + address);
+        Attributes attributes = dirContext.getAttributes(serviceDns, new String[]{"SRV"});
+        Attribute srvAttribute = attributes.get("srv");
+        if (srvAttribute != null) {
```

Comment:
I see we saved ourselves from NPE. What happens now when `srvAttribute` now? What is the path?

Reply:
The only change is in the logs. Before you had a stacktrace with NPE, now you get a message that the service name is not found. It happens when no POD (under the given service) is ready yet. So there is actually no functional change here, the discovery is repeated a few times and then gives up.